python/activator/middleware_interface.py

```diff
@@ -141,18 +174,32 @@ def __init__(self, central_butler: Butler, image_bucket: str, instrument: str,
         # How much to pad the refcat region we will copy over.
         self.padding = 30*lsst.geom.arcseconds
 
-    def _init_local_butler(self, butler: Butler):
+    def _make_apdb_uri(self):
+        """Generate a URI for accessing the APDB.
+        """
+        # TODO: merge this code with make_pgpass.py
```

Comment:
Is this something that could live in `analysis_ap`? There's code for attaching to an APDB in there, too.

Reply:
This is  just parsing the PP service's configuration (i.e., its envvars); what common ground would it have with `analysis_ap`? In any case, I don't think it's a good idea for the pipeline to depend on analysis packages.